roles/translator/proxy-config.toml

```diff
@@ -18,3 +18,18 @@ min_supported_version = 2
 
 # Minimum extranonce2 size
 min_extranonce2_size = 16
+
+# Difficulty params
+[downstream_difficulty_config]
+# hashes/s of the weakest miner that will be connecting
+min_individual_miner_hashrate=0.000_005
```

Comment:
> > so is the CPU miner not submitting at all or does it stop submitting after some amount of time? Also can you paste your `proxy-config.toml`?
> 
> It does submit shares. I added some details [above](https://github.com/stratum-mining/stratum/pull/474#discussion_r1126753395).
> 
> It's the upstream pool that hangs up. I don't think the translator submitted any of the shares upstream, presumably they don't meet the target:
> 
> ```
> Up: Updating Target to: Owned([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 128, 255, 127, 0, 0, 0, 0, 0])
> ```
> 
> I haven't calculated what that target translates to, but presumably much more. (see #479; it would be nice to have an easier to read message here)
> 
> ```shell
> $ cat ~/.stratum/proxy-config.toml 
> ```
> 
> ```toml
> # Braiins Pool Upstream Connection
> upstream_authority_pubkey = "u95GEReVMjK6k5YqiSFNqqTnKU4ypU2Wm8awa6tmbmDmk1bWt"
> upstream_address = "18.196.32.109"
> upstream_port = 3336
> 
> # Local Mining Device Downstream Connection
> downstream_address = "0.0.0.0"
> downstream_port = 63077
> 
> # Version support
> max_supported_version = 2
> min_supported_version = 2
> 
> # Minimum extranonce2 size
> min_extranonce2_size = 16
> 
> # Difficulty params
> [downstream_difficulty_config]
> # hashes/s of the weakest miner that will be connecting
> min_individual_miner_hashrate=0.0003
> # interval in seconds to elapse before updating the hashrate of a downstream connection
> miner_num_submits_before_update=5
> # target number of shares per minute the miner should be sending
> shares_per_minute = 2.0
> 
> [upstream_difficulty_config]
> # interval in seconds to elapse before updating channel hashrate with the pool
> channel_diff_update_interval = 60
> # estimated accumulated hashrate of all downstream miners
> channel_nominal_hashrate = 0.000_005
> ```

This looks like an extremely low target. Is this what braiins is sending with `SetTarget`?

Reply:
> The SV2 specifies nominal_hash_rate which is TH/s,

Ok, if the spec uses TH/s then it makes sense to use that everywhere.

Reply:
Oh I think you're referring to the `Up: Updating Target to:` log message, not my `channel_nominal_hashrate` config.

Reply:
pls use h/s everywhere. In the spec we use h/s not TH/s